### ISSUE TITLE
Chapter1 About Machine Learning 

### ISSUE THREAD

**Issue by yujingma45** [yujingma45/Blog#7]
Since machine learning is an important tools to solve problem, we need to ask following questions to understand it:

### What is machine learning?
Machine learning is used to solve complicated problems by **learning from data**, using a **performance measurement** to decide if a model can solve the model well.

### What is machine learning good at?
Machine learning is good at:
1. Make better decisions instead of writing down multiple complicated rules.
2. If a problem is under a keep changing environment, machine learning can help
3. Learn about a complex situation
4. There are a large dataset to deal with

### What is machine learning bad at?
Machine learning will behave badly when:
1. data is too small
2. poor quality data
3. data is no representative
4. data has some redundant features

### Machine learning types:
1. Separated by training task:
- Supervised vs. Unsupervised (Data has **label** or not; Label = data has a desired output or not)
- may also have semi-supervised and reinforcement learning 

- Supervised learning handles problems: 
   classification and regression
- Unsupervised learning will: 
   clustering, visualization, anomaly detection, associate rule and dimension reduction.

2. Separated by training **ways**:
- Online training 
(keep learning new data and discard old ones; easier to update and retrain; light weight; not only can be trained online but lightweight training)

- Batch training 
(Trained offline, data size big, hard and take longer to retrain)

- Out of core training
(not lightweight, but want to keep update, so using mini- batch + batch training to do so)

3. Separated by model **nature**:

- Instance-based prediction
(using each observation as part of the model. When predicting, calculating similarity distance between observations to make predictions)

- Model-based prediction
(using data set to tuning parameters in the model, and calculate the generalization error each time. When predicting, using model to make predictions)

### How to use machine learning:
1. Parameters vs. hyper-parameters (Not very clear)
Parameters - regarding to data
hyperparameter - regarding to the model/ learning algorithm itself

2. Training/ Test/ Validation data 
To avoid over-fitting and test the actual performance on the real problem, we need test data set to check the model performance. However, we cannot choose/ tune model based on the test data. If we do so, we actually also consider to fit test data into the model, then our real performance will be worse than the result on test data. So we need a validation data (separate from training data)  to choose/ tune the model.

To value the training dataset, we may use cross validation. (why we need cv? need more explanations)

3. Overfitting vs. Underfitting

Bias vs. variance trade off explanation - bias is the direction to shoot for a target (how to approach to the truth), variance is the average performance of the model (How well we can zoom)

To solve overfitting:
- Get more data
- Using simpler model
- Using less feature - feature selection
- Using regularization

To solve underfitting:
- Using more feature - feature engineering 
- Using complex model
